### DISCUSSION TITLE
Why am I getting this error about cork()?

### DISCUSSION THREAD

**Discussion by loliaravioli** [uNetworking/uWebSockets.js#1068] [Q&A] [answered]
I have this code:

```

const { App } = require('uWebSockets.js');
const app = new App();
const PORT = process.env.PORT || 5500;
app.any('/*', (res, req) => {
    const filePath = path.join(staticDir, req.getUrl() == '/' ? '/index.html' : req.getUrl());

    let fileContent,
        mimeType,
        status;

    try {
        fileContent = fs.readFileSync(filePath);
        mimeType = mime.lookup(filePath);
        status = '200 OK';
    } catch (err) {
        fileContent = 'File Not Found';
        mimeType = 'text/plain';
        status = '404 Not Found';
    }

    res.cork(() => {
        res.writeStatus(status)
            .writeHeader('Content-Type', mimeType)
            .end(fileContent);
    });
}).listen(PORT, function (token) {
    if (token) {
        console.log(`Server is running at http://localhost:${PORT}`);
    } else {
        console.log(`Failed to listen to port ${PORT}`);
    }
});
```

all it's supposed to be doing is serving static files, and it seems to be working, but I keep getting this warning:

```
Warning: uWS.HttpResponse writes must be made from within a corked callback. See documentation for uWS.HttpResponse.cork and consult the user manual.
```

As far as I can tell, I'm doing everything like the documentation says. The only time I'm writing a response is from within `cork()`. I managed to get rid of this warning temporarily by changing the `() => { }` to a `function() { }` but I had other errors and now it doesn't make a difference which one I use. Please help me figure out what I'm doing wrong or how to get rid of this warning.

**Comment by e3dio** [comment]
Your route is synchronous so `res.cork()` is not needed there, it is already corked. But you should change `readFileSync` to the async type so it doesn't block rest of app, then you need cork

**Reply by loliaravioli** [reply]
That makes sense... but (and I'm sorry to ask for some handholding) I'm still getting the same warnings.
```
const PORT = process.env.PORT || 5500;
app.any('/*', async (res, req) => {
    const filePath = path.join(staticDir, req.getUrl() == '/' ? '/index.html' : req.getUrl());

    res.onAborted(() => { console.warn('Request was aborted by the client'); });

    fs.readFile(filePath, (err, data) => {
        if (res.aborted) { return; }

        res.cork(() => {
            res.writeStatus(err ? '404 Not Found' : '200 OK')
                .writeHeader('Content-Type', err ? 'text/plain' : mime.lookup(filePath))
                .end(err ? 'File Not Found' : data);
        });
    });
}).listen(PORT, function (token) {
    if (token) {
        console.log(`Server is running at http://localhost:${PORT}`);
    } else {
        console.log(`Failed to listen to port ${PORT}`);
    }
});
```

Though this is definitely more correct than what I had so thank you for that.

**Reply by e3dio** [reply]
Is this the only code you are running, or are you hiding bunch of other code that could have error? You are supposed to have `res.onAborted(() => res.aborted = true)` but that would give different error

**Reply by loliaravioli** [reply]
This is the entirety of the server code. I commented out everything else.

The warnings seem to only trigger when I do a Ctrl+Shift+R (hard refresh). Why that would be, I don't know. It refreshes some CSS but I doubt that's the issue.

I'll add in the aborted variable. Thank you for pointing that out.

**Reply by e3dio** [reply, marked as answer]
You are trying to combine Socket.io with uWS.js here https://github.com/loliaravioli/web-hearthstone/blob/main/server.js

That is very likely giving the warning. Why did you leave that out from your report ??? Why would you use Socket.io ???? You don't need that, use uWS.js by itself

**Reply by e3dio** [reply]
> This is the entirety of the server code

Except for the part where you tried to combine uWS.js and Socket.io into a Frankenstein monster app, and you hid that from your post so I have to go dig thru your repos to find what you are trying to do

**Reply by loliaravioli** [reply]
I left it out because I thought I had commented out `io.attachApp(app)` when I was testing but I guess I made a mistake. Sorry for wasting your time.

As for why I'm using socket.io... it's convenient and more concise than uWS.js. I'm using socket.io for its high-level API and uWS.js for its lower-level performance benefits.

**Reply by e3dio** [reply]
Clearly it is not properly supported because it is giving you warnings. Do yourself a favor and get rid of Socket.io and use uWS.js, much nicer to use, better API, better performance, etc. Can't help you if you use Socket.io

**Reply by loliaravioli** [reply]
I appreciate the warning but I'm getting the uWS.js performance already. I'll live with a couple of warnings on server startup.

**Reply by e3dio** [reply]
Ok please open issues in Socket.io not here
